### PR TITLE
Test for left menu panel

### PR DESCRIPTION
**Pull request by Pulower**
Check if it's ok

### CODE REVIEW

File: src/test/java/base/TestBase.java [hunk 1]
```diff
@@ -37,24 +38,26 @@ private static void printBrowserLogs(WebDriver driver) throws IOException {
...
     }
 
     @BeforeTest
-    public void setUpDriver(){
+    public void setUpDriver() {
         driverManager = DriverManagerFactory.getManager(DriverType.CHROME);
         driver = driverManager.getDriver();
         driver.get(address);
         driver.manage().window().maximize();
+        LeftNavigationPanelPage leftPanel = new LeftNavigationPanelPage(driver);
```
Comment (marcinberus) [inline src/test/java/base/TestBase.java]: this should be in particular test, not before every test :)

File: src/test/java/base/TestBase.java [hunk 2]
```diff
@@ -37,24 +38,26 @@ private static void printBrowserLogs(WebDriver driver) throws IOException {
...
+        leftPanel.acceptCookies();
```
Comment (marcinberus) [inline src/test/java/base/TestBase.java]: same

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+        try {
+            news.isDisplayed();
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    public void checkPromotions() {
+        clickOnElement(promotions);
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: instructions in method miss its name, should be just clickPromotion()

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+    }
+
+    public void checkBestsellers() {
+        clickOnElement(bestsellers);
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: same

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+    }
+
+    public void checkNews() {
+        clickOnElement(news);
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: same

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+            case ZAMOW_SKLEP_INTERNETOWY: {
+                clickOnElement(internetShop);
+                break;
+            }
+        }
+        return this;
+    }
+
+    public void acceptCookies() {
+        clickOnElement(cookies);
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: cookies is only with left nav page? ;)

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 7]
```diff
@@ -0,0 +1,390 @@
...
+    }
+
+    public boolean isInPromotion() {
+        try {
+            promotions.isDisplayed();
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: what you think about:
public boolean isInPromotion() {
return promotions.isDisplayed();
}

Comment (marcinberus) [reply src/main/java/Pages/LeftNavigationPanelPage.java]: but also you need getter:

public WebElement getPromotion() {
waitUntilElementIsVisible(promotion);
return promotion;
}

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 8]
```diff
@@ -0,0 +1,390 @@
...
+
+    public boolean isBestseller() {
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: same

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 9]
```diff
@@ -0,0 +1,390 @@
...
+            bestsellers.isDisplayed();
+            return true;
+        } catch (NoSuchElementException e) {
+            return false;
+        }
+    }
+
+    public boolean isNew() {
+        try {
+            news.isDisplayed();
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: same

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 10]
```diff
@@ -0,0 +1,390 @@
...
+    private WebElement ebooksAndAudiobooks;
+
+    @FindBy(partialLinkText = "ebooki")
+    private WebElement ebooks;
+
+    @FindBy(partialLinkText = "audiobooki")
+    private WebElement audiobooks;
+
+    @FindBy(partialLinkText = "Zamów sklep internetowy")
+    private WebElement internetShop;
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: when i type Zamów sklep internetowy or Sklep RTV in devtools i can find 2 and 3 'a href's. i want particular element with link in left nav page :) cwaniaku

File: src/main/java/Pages/LeftNavigationPanelPage.java [hunk 11]
```diff
@@ -0,0 +1,390 @@
...
+import org.openqa.selenium.support.FindBy;
+
+public class LeftNavigationPanelPage extends WebElementManipulator {
+
+    public LeftNavigationPanelPage(WebDriver driver) {
+        super(driver);
+    }
+
+    @FindBy(partialLinkText = "Promocje")
+    private WebElement promotions;
```
Comment (marcinberus) [inline src/main/java/Pages/LeftNavigationPanelPage.java]: next comment tells you what to do with **** partialtext handle

**Comment by Pulower** [comment]
It can't be, because when i use isDisplayed method on element that doesn't
exist at the moment, it will throw an exception.

2018-03-15 21:55 GMT+01:00 Marcin Berus <notifications@github.com>:

**Pull request closed without merging**